test/unittests/execute_floating_point_test.cpp

```diff
@@ -417,3 +417,229 @@ TYPED_TEST(execute_floating_point_trunc, trunc)
         EXPECT_EQ(result.value.template as<IntT>(), FloatT{-1});
     }
 }
+
+TEST(execute_floating_point, f32_load)
+{
+    /* wat2wasm
+    (memory 1 1)
+    (data (i32.const 0) "\00\00\00\00") ;; 0
+    (data (i32.const 4) "\b6\f3\9d\3f") ;; 1.234
+    (data (i32.const 8) "\b6\f3\9d\bf") ;; -1.234
+    (data (i32.const 12) "\00\00\c0\7f") ;; canonical NaN
+    (data (i32.const 16) "\01\00\c0\7f") ;; arithmetic NaN
+    (data (i32.const 20) "\01\00\80\7f") ;; signaling NaN
+    (func (param i32) (result f32)
```

Comment:
Can you add all the representations from `floating_point_utils`, i.e. include infinity?

Reply:
Added -0 and infinities, also to store tests

Reply:
Don't want to stretch it too much, but would it make sense having every case form` floating_point_utils:double_as_uint`, etc?

Reply:
More cases would ask for a general table approach... I'm inclined to say it's not worth it

Reply:
> Don't want to stretch it too much, but would it make sense having every case form` floating_point_utils:double_as_uint`, etc?

These values are not very specific. As it is is fine to me.

Reply:
I think they are nice examples though, and shows if any part (beginning or end) is cut from loading/storing.